### PR TITLE
[dbus] fix OTBR's behavior when `Reset` is invoked

### PR DESCRIPTION
**Pull request by superwhd**
When the D-BUS API `Reset` is invoked, 

- re-register `UpdateMeshCopTxtHandler`
- reset advertising proxy callbacks
- reset discovery proxy callbacks

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

Thanks

File: tests/dbus/test-client [hunk 1]
```diff
@@ -79,6 +79,14 @@ update_meshcop_txt_and_check()
...
     grep --binary-files=text "A=abc" <<<"${service}"
     grep --binary-files=text "B=12" <<<"${service}"
+
+    sudo gdbus call --system --dest io.openthread.BorderRouter.wpan0 --method=io.openthread.BorderRouter.Reset --object-path /io/openthread/BorderRouter/wpan0
+    sleep 30
+
+    sudo gdbus call --system --dest io.openthread.BorderRouter.wpan0 --method=io.openthread.BorderRouter.UpdateVendorMeshCopTxtEntries --object-path /io/openthread/BorderRouter/wpan0 "[('C',[97,98,99])]"
+    sleep 5
+    service="$(scan_meshcop_service)"
+    grep --binary-files=text "C=abc" <<<"${service}"
```
Comment (jinran-google) [inline tests/dbus/test-client:89]: Shall we also add tests for discovery proxy?

**Review by jinran-google — APPROVED** [review]
LGTM overall.

File: tests/dbus/test-client (hunk 1 above)
Comment (superwhd) [reply tests/dbus/test-client]: Currently this fix is kind of flaky with mDNSResponder. I'll add the test cases later in another PR.

**Review by jwhui — APPROVED** [review]

**Pull request merged**